### ISSUE TITLE
No EPG 

### ISSUE THREAD

**Issue by pejobo** [rsiebert/TVHClient#231]
After updating server (tvheadend version 4.1-1870~g1cc14cd), no EPG data is shown in TVHClient. I sent you a log (from advanced settings menu), can't find it on the device to include it here.
The web interface of tvheadend is showing the epg data..

**Comment by jvloendersloot** [comment]
Same problem here. Using tvheadend version 4.1-1929~g2a70a90. Can't find any logfiles either.

/edit

Just upgraded to version 4.1-1931~gc3eefc6. Same problem. I see my tv logo's and channel names, but no program information.

**Comment by rsiebert** [comment]
The last change on the htsp protocol was on the 13.4.2016 (See https://github.com/tvheadend/tvheadend/commits/master/src/htsp_server.c). Is your server older or newer than that? Maybe the commit "htsp server: fix the previous commit - getEvents access verification fix" fixed the EPG issue.

The logfile is saved in the cache folder of the app.

**Comment by pejobo** [comment]
My version is older (one day). But Justins version is newer (so should contain the commit):

```
$ git log --since=3.weeks | grep -A 4 1cc14cd # my version 
commit 1cc14cdb2f3faa6ec652b67de406b1cc46ab739e
Author: Jaroslav Kysela <perex@perex.cz>
Date:   Mon Apr 11 18:25:04 2016 +0200

    DVR: try to fix missing channel assignments

$ git log --since=3.weeks | grep -A 4 2a70a90 # version of jvloendersloot
commit 2a70a903e2a68681f45254a1a0297b25700d40b9
Author: Jaroslav Kysela <perex@perex.cz>
Date:   Thu Apr 21 13:41:16 2016 +0200

    eit: always register mux for uk_freesat services, fixes #3740

$ git log --since=3.weeks | grep -A 4 37ed360 # the 2nd commit you mentioned
commit 37ed360d1ab694554489d18071fc3c529d0d2d4f
Author: Jaroslav Kysela <perex@perex.cz>
Date:   Mon Apr 11 11:35:06 2016 +0200

    htsp server: fix the previous commit - getEvents access verification fix

```

**Comment by pejobo** [comment]
I tested yesterday evening with tvheadend config. With a new user (same permissions as default user) the epg is displayed. 
I'll try to dig deeper this evening..

**Comment by pejobo** [comment]
Wrong finding. EPG is displayed (sort of) after a search is issued.

**Comment by rsiebert** [comment]
I updated the server to HTS Tvheadend 4.1-1933~g7f24603 from two days ago. With the official version 1.9.1 and the beta version 1.9.2 I see all programs. Strange.

**Comment by pejobo** [comment]
No change after update to tvheadend-git-4.1.r1943.g10f1326-1. Webinterface and Kodi show EPG fine. Maybe a config issue somewhere?

**Comment by pejobo** [comment]
I re-checked the config and found no error. I also tried another client (iOS), which worked flawlessly.
On TVHClient the epg data is only displayed after I issue a search. And it disappears again after some time. Before it disappears the display gets out of sync.
Recordings and recording schedule is displayed correct. Streaming is also working.

**Comment by rsiebert** [comment]
Another user had the same issue and reported that it has something to do with the access rights. The initial admin user works but not any new users. Then he was able to pinpoint the problem.
If he wanted to give the new user other rights, no epg will be shown. For example restrikting the "minimal channel number" and "maximal channel number" or "channel tags".

![image](https://cloud.githubusercontent.com/assets/1406960/15037249/927315c4-1299-11e6-8bee-095397d933ab.png)

**Comment by rsiebert** [comment]
With my latest tvheadend version HTS Tvheadend 4.1-1941 I also did not get eny epg data. After reverting back to 4.1-1354 I get it again. So this looks more and more to a server issue to me. The HTSP API also did not change in the newer versions.

**Comment by pejobo** [comment]
I also believe that something on the server side has changed. But on the other hand I've got two other software clients which display the epg still fine..

**Comment by pejobo** [comment]
Perhaps the other user you mentioned can provide a working config file content?

**Comment by satip** [comment]
i opened a case:
https://tvheadend.org/issues/3808

**Comment by perexg** [comment]
Fixed in v4.1-2044-g0a2958a .

**Comment by rsiebert** [comment]
Thanks for creating an issue and helping resolving this problem!

**Comment by satip** [comment]
tested and fixed with version HTS Tvheadend 4.1-2045~gf027e3c

**Comment by jvloendersloot** [comment]
I can confirm with version 4.1-2070~g9407f1b it's working again.
